src/main/java/mate/academy/rickandmorty/service/impl/CharacterServiceImpl.java

```
@@ -0,0 +1,30 @@
+package mate.academy.rickandmorty.service.impl;
+
+import java.util.List;
+import lombok.RequiredArgsConstructor;
+import mate.academy.rickandmorty.dto.CharacterDto;
+import mate.academy.rickandmorty.mapper.CharacterMapper;
+import mate.academy.rickandmorty.model.Character;
+import mate.academy.rickandmorty.repository.CharacterRepository;
+import mate.academy.rickandmorty.service.CharacterService;
+import org.springframework.stereotype.Service;
+
+@RequiredArgsConstructor
+@Service
+public class CharacterServiceImpl implements CharacterService {
+    private final CharacterRepository characterRepository;
+    private final CharacterMapper characterMapper;
+
+    @Override
+    public CharacterDto getRandomCharacter() {
+        return characterMapper.toDto(characterRepository.getRandomCharacter());
```

While using a native query with `ORDER BY RAND()` prevents the application from crashing on an empty database, it's not the most robust way to handle this scenario. If no characters exist, `getRandomCharacter()` will return `null`, and the API will respond with a 200 OK and an empty body. 

A better approach would be to check if the database contains any characters (e.g., with `characterRepository.count()`). If it's empty, you could throw a specific exception, allowing your controller to return a more informative response, like a 404 Not Found.